ion-hash/src/representation.rs

```diff
@@ -66,36 +82,45 @@ fn write_repr_integer<U: Update>(value: Option<&AnyInt>, hasher: &mut UpdateEsca
     }
 }
 
+/// Ion hash defines representations for some special values.
+struct Floats;
+impl Floats {
+    const NEGATIVE_ZERO: [u8; 8] = [0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00];
+    const POSITIVE_INFINITY: [u8; 8] = [0x7F, 0xF0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00];
+    const NEGATIVE_INFINITY: [u8; 8] = [0xFF, 0xF0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00];
+    const NAN: [u8; 8] = [0x7F, 0xF8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00];
+}
```

Comment:
I don't mind the namespacing, but I'd be cool with just keeping them flat in the module.